osu.Game/Screens/Play/HUD/SoloGameplayLeaderboard.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Users;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public class SoloGameplayLeaderboard : GameplayLeaderboard
+    {
+        private readonly IUser trackingUser;
+
+        public readonly IBindableList<ScoreInfo> Scores = new BindableList<ScoreInfo>();
+
+        // hold references to ensure bindables are updated.
+        private readonly List<Bindable<long>> scoreBindables = new List<Bindable<long>>();
+
+        [Resolved]
+        private ScoreProcessor scoreProcessor { get; set; } = null!;
+
+        [Resolved]
+        private ScoreManager scoreManager { get; set; } = null!;
+
+        public SoloGameplayLeaderboard(IUser trackingUser)
+        {
+            this.trackingUser = trackingUser;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Scores.BindCollectionChanged((_, _) => Scheduler.AddOnce(showScores), true);
```

Comment:
When finishing gameplay, the user's submitted online score seems to be added to the `Scores` list, which makes it appear on the leaderboard:

https://user-images.githubusercontent.com/22781491/191360700-d526ed6b-148c-4089-bd09-d3efbd4114d9.mp4

I think that could be fixed by checking against the score ID, similar to how it's done on results screens.

Reply:
This is a bit unfortunate, as it was avoided by having the schedule in place previously.

I've fixed this by unbinding the scores, rather than trying any kind of ID matching (sounds awkward).